src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -340,7 +340,6 @@ public override void Dispose()
                     _webSocketReceiveResultTcs = new TaskCompletionSource<WebSocketReceiveResult>();
                 }
 
-                InterlockedCheckValidStates(s_validAfterReceiveStates);
```

Comment:
> Do we have anything similar in the other implementations that we need to fix similarly? Or maybe this was already removed from those (if it was ever there)?

The other implementation has a different approach for handling callback failures which requires this check. A recent change in the WinRT implementation added a different way but this check was not removed, causing the failure in corner-case race conditions between operations or when the callback is called because of a server RST or malformed ws frame.